src/main/java/ca/corefacility/bioinformatics/irida/processing/impl/AutomatedAnalysisFileProcessor.java

```diff
@@ -0,0 +1,185 @@
+					.build();
+
+			submission = submissionRepository.save(submission);
+
+			//share submission back to the project
+			Project project = template.getSubmittedProject();
+			pasRepository.save(new ProjectAnalysisSubmissionJoin(project, submission));
+
+			legacyFileProcessorCompatibility(submission, sequencingObject);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Boolean modifiesFile() {
+		return false;
+	}
+
+	/**
+	 * Get the {@link AnalysisSubmissionTemplate}s for a given {@link SequencingObject}.  Search up the {@link Sample}
+	 * and {@link Project}s the sequence belongs to.
+	 *
+	 * @param object the {@link SequencingObject}
+	 * @return the List of {@link AnalysisSubmissionTemplate}
+	 */
+	private List<AnalysisSubmissionTemplate> getAnalysisTemplates(SequencingObject object) {
+		List<AnalysisSubmissionTemplate> submissionTemplates = new ArrayList<>();
+
+		SampleSequencingObjectJoin sampleForSequencingObject = ssoRepository.getSampleForSequencingObject(object);
+
+		/*
+		 * Checking if the seq object was deleted from the sample before this file processor was run
+		 */
+		if (sampleForSequencingObject != null) {
+			List<Join<Project, Sample>> projectForSample = psjRepository.getProjectForSample(
+					sampleForSequencingObject.getSubject());
+
+			//get all the projects for the sample
+			for (Join<Project, Sample> j : projectForSample) {
+
+				//get the analysis templates for that project
+				List<AnalysisSubmissionTemplate> analysisSubmissionTemplatesForProject = analysisTemplateRepository.getAnalysisSubmissionTemplatesForProject(
+						j.getSubject());
+
+				//check if the project owns this sample
+				ProjectSampleJoin psj = (ProjectSampleJoin) j;
+				boolean owner = psj.isOwner();
+
+				analysisSubmissionTemplatesForProject.forEach(t -> {
+					//adding the sample name to the template
+					String name = t.getName();
+					name = name + " - " + j.getObject()
+							.getSampleName();
+					t.setName(name);
+
+					//don't try to update the sample if this project isn't the owner.  it'll fail when it tries.
+					if (!owner) {
+						t.setUpdateSamples(false);
+					}
+				});
+
+				submissionTemplates.addAll(analysisSubmissionTemplatesForProject);
+			}
+
+		} else {
+			logger.warn("Cannot find sample for sequencing object " + object.getId()
+					+ ".  Not running automated pipelines.");
+		}
+
+		return submissionTemplates;
+	}
+
+	/**
+	 * Do the work the old Assembly and SISTR file processors used to do and assign the assembly and SISTR result back
+	 * to the {@link SequencingObject}.  This will only do something if its an Assembly or SISTR analysis.
+	 *
+	 * @param submission       the {@link AnalysisSubmission} to check
+	 * @param sequencingObject the {@link SequencingObject} to apply results to.
+	 */
+	private void legacyFileProcessorCompatibility(AnalysisSubmission submission, SequencingObject sequencingObject) {
```

Comment:
Similar to skipping `updateSamples` if the sample is locked in the project, I think the code to link the analysis submission to the sequencing object should also be skipped if the particular sequencing object is locked in the project.

Reply:
Good catch.  Done in 050cf5e